File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/MemberDTO.java
package com.ll.hotel.domain.member.member.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;

import com.ll.hotel.domain.member.member.entity.Role;
import com.ll.hotel.domain.member.member.type.MemberStatus;
import com.ll.hotel.domain.member.member.entity.Member;

public record MemberDTO(
        Long id,
        String memberEmail,
        String memberName,
        String memberPhoneNumber,
        LocalDate birthDate,
        LocalDateTime createdAt,
        LocalDateTime modifiedAt,
        Role role,
        MemberStatus memberStatus
) {
    public static MemberDTO from(Member member) {
        return new MemberDTO(
            member.getId(),
            member.getMemberEmail(),
            member.getMemberName(),
            member.getMemberPhoneNumber(),
            member.getBirthDate(),
            member.getCreatedAt(),
            member.getModifiedAt(),
            member.getRole(),
            member.getMemberStatus()
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/MemberDTO.kt
package com.ll.hotel.domain.member.member.dto

import java.time.LocalDate
import java.time.LocalDateTime

import com.ll.hotel.domain.member.member.entity.Role
import com.ll.hotel.domain.member.member.type.MemberStatus
import com.ll.hotel.domain.member.member.entity.Member

data class MemberDTO(
    val id: Long,
    val memberEmail: String,
    val memberName: String,
    val memberPhoneNumber: String,
    val birthDate: LocalDate?,
    val createdAt: LocalDateTime,
    val modifiedAt: LocalDateTime,
    val role: Role,
    val memberStatus: MemberStatus
) {
    companion object {
        @JvmStatic
        fun from(member: Member): MemberDTO {
            return MemberDTO(
                id = member.id,
                memberEmail = member.memberEmail,
                memberName = member.memberName,
                memberPhoneNumber = member.memberPhoneNumber,
                birthDate = member.birthDate,
                createdAt = member.createdAt,
                modifiedAt = member.modifiedAt,
                role = member.role,
                memberStatus = member.memberStatus
            )
        }
    }
} 